File: university/src/main/java/com/gutorov/university/api/application/ApplicationRs.java
package com.gutorov.university.api.application;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.gutorov.university.api.program.ProgramRs;
import com.gutorov.university.entity.ProgramEntity;

import java.time.LocalDateTime;
import java.util.UUID;

public class ApplicationRs {
    private UUID id;
    private String fullName;
    private String email;
    private ProgramRs program;
    private LocalDateTime submissionDate;
    @JsonProperty("isAdmitted")
    private boolean isAdmitted;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public ProgramRs getProgram() {
        return program;
    }

    public void setProgram(ProgramRs program) {
        this.program = program;
    }

    public LocalDateTime getSubmissionDate() {
        return submissionDate;
    }

    public void setSubmissionDate(LocalDateTime submissionDate) {
        this.submissionDate = submissionDate;
    }

    public boolean isAdmitted() {
        return isAdmitted;
    }

    public void setAdmitted(boolean isAdmitted) {
        this.isAdmitted = isAdmitted;
    }
}

File: src/main/java/com/gutorov/university/api/application/ApplicationRs.kt
package com.gutorov.university.api.application

import com.fasterxml.jackson.annotation.JsonProperty
import com.gutorov.university.api.program.ProgramRs
import com.gutorov.university.entity.ApplicationEntity
import java.time.LocalDateTime
import java.util.*
import java.util.stream.StreamSupport

class ApplicationRs(
    var id: UUID? = null,
    var fullName: String? = null,
    var email: String? = null,
    var program: ProgramRs? = null,
    var submissionDate: LocalDateTime? = null,
    @JsonProperty("isAdmitted")
    var isAdmitted: Boolean = false
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: ApplicationEntity): ApplicationRs {
            return ApplicationRs(
                entity.id,
                entity.fullName,
                entity.email,
                ProgramRs.fromEntity(entity.program),
                entity.submissionDate,
                entity.isAdmitted
            )
        }

        @JvmStatic
        fun fromEntityList(entities: List<ApplicationEntity>): List<ApplicationRs> {
            return StreamSupport.stream(entities.spliterator(), false)
                .map(ApplicationRs::fromEntity)
                .toList();
        }
    }
}
